src/pkg/services/m365/api/lists.go

```diff
@@ -490,20 +518,57 @@ func setAdditionalDataByColumnNames(
 	fieldData := orig.GetAdditionalData()
 	filteredData := make(map[string]any)
 
-	for colName, colDetails := range columnNames {
-		if val, ok := fieldData[colName]; ok {
+	for _, colDetails := range columnNames {
+		if val, ok := fieldData[colDetails.getFieldName]; ok {
 			setMultipleEnabledByFieldData(val, colDetails)
-
-			filteredData[colName] = fieldData[colName]
+			filteredData[colDetails.createFieldName] = val
+			populateMultipleValues(val, filteredData, colDetails)
 		}
 
-		specifyODataType(filteredData, colDetails, colName)
+		specifyODataType(filteredData, colDetails, colDetails.createFieldName)
 	}
 
 	return filteredData
 }
 
+func populateMultipleValues(val any, filteredData map[string]any, colDetails *columnDetails) {
+	if !colDetails.isMultipleEnabled {
```

Comment:
nit: `isMultipleEnabled` and `isPersonColumn` checks can be folded into a single if block

Reply:
yeah. In next PR I was going to add for lookupColumn check. Hence wanted to keep checks separate for their corresponding reasoning.